app/Platform/Concerns/ActsAsPlayer.php

```diff
@@ -39,6 +40,18 @@ public function rollConnectToken(): void
         $this->save();
     }
 
+    // == instance methods
+
+    public function hasPlayed(Game $game): bool
+    {
+        return $this->playerGame($game)->exists();
```

Comment:
is it safe to call `->exists()` on `null`?

Reply:
For full disclosure, `hasPlayed()` already existed on the model. In this PR, I relocated it in the file (it was listed in the "relations" section but is more of an internal method than a relation).

However, your observation is astute. No, this usage of `->exists()` is not safe.

I've rewritten `hasPlayed()`:

```php
public function hasPlayed(Game $game): bool
{
    return $this->playerGames()->where('game_id', $game->id)->exists();
}
```